File: LD22/javascript/sound.js

var volume = 0.5
var sounds = {}

function play(name, delay) {
    if (!volume) return
    if (!sounds[name]) {
        sounds[name] = new Array(10)
        // if there's a sound you need to play more than 10 of at a time, cry me a river
        for (var j = 0 ; j < 10 ; ++j) sounds[name][j] = new Audio("sound/" + name + ".ogg")
        sounds[name].next = 0
    }
    var sound = sounds[name][sounds[name].next]
    sounds[name].next = (sounds[name].next + 1) % 10
    sound.volume = volume
    if (delay) {
        setTimeout(sound.play, delay)
    } else {
        sound.play()
    }
}

var musicvolume = 0.3
var music = null
var musicname = null
function playmusic(name) {
    if (musicname == name) return // already playing
    musicname = name
    if (music) music.stop()
    music = new Audio("music/" + name + ".ogg")
    music.loop = true
    music.volume = musicvolume
    music.play()
}

exports.play = play
exports.playmusic = playmusic

File: last-will/src/sound.py
import pygame
import data, settings, menu

sounds = {}
def play(soundname):
	if not settings.audio or not settings.sound:
		return
	if menu.stack:
		return
	if soundname not in sounds:
		sounds[soundname] = pygame.mixer.Sound(data.filepath("sound", "%s.ogg" % soundname))
	sounds[soundname].play()

currentmusic = None
def playmusic(songname):
	global currentmusic
	if not settings.audio or not settings.music:
		return
	if songname == currentmusic:
		return
	currentmusic = songname
	pygame.mixer.music.stop()
	pygame.mixer.music.load(data.filepath("music", "%s.ogg" % songname))
	pygame.mixer.music.play(-1)


